### DISCUSSION TITLE
Access to unmodified `Annotated` in v1 (v2 has `metadata` flag)

### DISCUSSION THREAD

**Discussion by DanCardin** [pydantic/pydantic#8181] [Question] [answered]
Asking as someone trying to provide backward compatibility with a library written for pydantic v2, back to pydantic v1

For some base model

```python
class Model(BaseModel):
    foo: Annotated[str, Something()]
```
`Model.model_fields['name']` yields a `ModelField` object that has `annotation` and `metadata` fields in pydantic v2; which would be how to access `str` and `Something()` respectively.

pydantic v1 has `annotation`, but not `metadata`, and it's unclear to me whether it is possible to access whatsoever. I dont see anything obvious in the resultant class, and searching the issue tracker has been fruitless thus far as many things include "annotated" that have nothing to do with `Annotated`

**Comment by tachikoma-li** [comment, marked as answer]
IIUC, you could use the `__annotations__` attribute, which returns a `typing.Annotated` and then the `__metadata__` would give the `metadata` as explained [here](https://docs.python.org/3/library/typing.html#typing.Annotated). That is, `Model.__annotations__["foo"].__metadata__` might be what you need.

**Reply by DanCardin** [reply]
mmm. so from my own tests, `__annotations__` doesn't work because it returns all string annotations (at least in some of cases).

With that said, I just realized that calling `get_type_hints` on the class still works, so I can map fields to that and handle it manually, I suppose.

**Reply by tachikoma-li** [reply]
Interesting, maybe in that case, `inspect.get_annotations` could work (and is preferred over  over `__annotations__` in general) as suggested [here](https://docs.python.org/3/howto/annotations.html#manually-un-stringizing-stringized-annotations).

**Reply by DanCardin** [reply]
It's just been the case that all of dataclasses, pydantic v2, pydantic's dataclasses, and attrs produce a per-field attribute that contains the annotation, including `Annotated`.

Whereas v1 does not, and given this, am finding I need to do the extra `get_type_hints` bit. I think I do in fact want the extra stuff that `get_type_hints` does. Perhaps that means I should be using it across the board...

In any case, while perhaps your suggestion wasn't the whole answer, it definitely immediately led me to find what appears to be the answer. I had definitely written v1 off and assumed it couldn't be supported. so for that I thank you!
